### PR TITLE
Fix _STD case of #825

### PR DESCRIPTION
**Pull request by AlexGuteniev**
- Detect use of `std` instead of `::std`
- Omit:
  - for `<future>` due to `<ppltask.h>`
  -  for `<new.h>` since it is not a part of STL
- Include `<vcruntime_new.h>` in ahead due to `std::nothrow_t` and `std::align_val_t`

### CODE REVIEW

File: tests/std/tests/Dev11_0000000_include_each_header_alone/test.cpp [hunk 1]
```diff
@@ -1,6 +1,15 @@
 // Copyright (c) Microsoft Corporation.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+#ifndef MEOW_DONT_FAKE_STD
+
+#include <vcruntime_new.h>
```
Comment (CaseyCarter) [inline tests/std/tests/Dev11_0000000_include_each_header_alone/test.cpp:6]: This unfortunately damages the original purpose of this test, which is to guarantee that each header can be included with no prerequisites. I'd prefer to see a separate/new test. Also, Stephan mentioned that there doesn't seem to be any advantage to doing this with each header individually and we could just as well make it an "include all the headers at once" test with `__msvc_all_public_headers.hpp`.

Comment (AlexGuteniev) [reply tests/std/tests/Dev11_0000000_include_each_header_alone/test.cpp:6]: If you can fix `<runtime_new>`, we can combine. Advantage is maintaining full list in one place

Comment (AlexGuteniev) [reply tests/std/tests/Dev11_0000000_include_each_header_alone/test.cpp:6]: Ah, you already have `__msvc_all_public_headers.hpp`

Comment (AlexGuteniev) [reply tests/std/tests/Dev11_0000000_include_each_header_alone/test.cpp]: Reverted here, added new implementation.

**Comment by AlexGuteniev** [comment]
@CaseyCarter , I don't get why some of runs pass and other fail in `<experimental/vector>`. Could be compiler bug. Will it be investigated, or we'll just close this PR?

**Comment by AlexGuteniev** [comment]
Here is end of my local test run, notice some pass and some fail:

```
ASS: std :: tests/GH_000825_inner_std_namespace:26 (23 of 29)
stl-lit.py: C:\Project\STL\tests\utils\stl\test\format.py:203: note: Build step failed unexpectedly.
Command: "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.26.28801\bin\HostX64\x64\cl.EXE" "/FeC:\Project\STL\out\build\x64\tests\std\tests\GH_000825_inner_std_namespace\18\GH_000825_inner_std_namespace.exe" "/FoC:\Project\STL\out\build\x64\tests\std\tests\GH_000825_inner_std_namespace\18\GH_000825_inner_std_namespace.obj" "/IC:\Project\STL\out\build\x64\out\inc" "/IC:/Project/STL/llvm-project/libcxx/test/support" "/IC:/Project/STL/tests/std/include" "/nologo" "/Od" "/W4" "/w14061" "/w14242" "/w14265" "/w14365" "/w14582" "/w14583" "/w14587" "/w14588" "/w14749" "/w14841" "/w14842" "/w15038" "/w15214" "/w15215" "/w15216" "/w15217" "/sdl" "/WX" "/Zc:strictStrings" "/D_ENABLE_STL_INTERNAL_CHECK" "/D_ENFORCE_FACET_SPECIALIZATIONS=1" "/bigobj" "/Za" "/EHsc" "/MDd" "/std:c++latest" "/permissive-" "/w14640" "/Zc:threadSafeInit-" "C:\Project\STL\tests\std\tests\GH_000825_inner_std_namespace\test.cpp" "/link" "/LIBPATH:C:\Project\STL\out\build\x64\out\lib\amd64" "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.26.28801\lib\x64"
Exit Code: 2
Standard Output:
--
test.cpp
C:\Project\STL\tests\std\tests\GH_000825_inner_std_namespace\test.cpp(17): error C2872: 'std': ambiguous symbol
C:\Project\STL\out\build\x64\out\inc\experimental/vector(21): note: could be 'std'
C:\Project\STL\tests\std\tests\GH_000825_inner_std_namespace\test.cpp(10): note: or       'X::std'
--


FAIL: std :: tests/GH_000825_inner_std_namespace:18 (24 of 29)
PASS: std :: tests/GH_000825_inner_std_namespace:17 (25 of 29)
PASS: std :: tests/GH_000825_inner_std_namespace:27 (26 of 29)
PASS: std :: tests/GH_000825_inner_std_namespace:28 (27 of 29)
stl-lit.py: C:\Project\STL\tests\utils\stl\test\format.py:203: note: Build step failed unexpectedly.
Command: "C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.26.28801\bin\HostX64\x64\cl.EXE" "/analyze:logC:\Project\STL\out\build\x64\tests\std\tests\GH_000825_inner_std_namespace\16\GH_000825_inner_std_namespace.nativecodeanalysis.xml" "/IC:\Project\STL\out\build\x64\out\inc" "/IC:/Project/STL/llvm-project/libcxx/test/support" "/IC:/Project/STL/tests/std/include" "/nologo" "/Od" "/W4" "/w14061" "/w14242" "/w14265" "/w14365" "/w14582" "/w14583" "/w14587" "/w14588" "/w14749" "/w14841" "/w14842" "/w15038" "/w15214" "/w15215" "/w15216" "/w15217" "/sdl" "/WX" "/Zc:strictStrings" "/D_ENABLE_STL_INTERNAL_CHECK" "/D_ENFORCE_FACET_SPECIALIZATIONS=1" "/bigobj" "/EHsc" "/MTd" "/D_ITERATOR_DEBUG_LEVEL=2" "/std:c++latest" "/analyze:only" "/w14640" "/Zc:threadSafeInit-" "C:\Project\STL\tests\std\tests\GH_000825_inner_std_namespace\test.cpp"
Exit Code: 2
Standard Output:
--
test.cpp
C:\Project\STL\tests\std\tests\GH_000825_inner_std_namespace\test.cpp(17): error C2872: 'std': ambiguous symbol
--


FAIL: std :: tests/GH_000825_inner_std_namespace:16 (28 of 29)
PASS: std :: tests/GH_000825_inner_std_namespace:11 (29 of 29)
********************
Failing Tests (8):
  std :: tests/GH_000825_inner_std_namespace:05
  std :: tests/GH_000825_inner_std_namespace:06
  std :: tests/GH_000825_inner_std_namespace:09
  std :: tests/GH_000825_inner_std_namespace:13
  std :: tests/GH_000825_inner_std_namespace:14
  std :: tests/GH_000825_inner_std_namespace:15
  std :: tests/GH_000825_inner_std_namespace:16
  std :: tests/GH_000825_inner_std_namespace:18


Testing Time: 7.60s
  Unsupported Tests  :  7
  Expected Passes    : 14
  Unexpected Failures:  8

1 warning(s) in tests

```

**Comment by StephanTLavavej** [comment]
Oh yeah - I remember filing a compiler bug a long time ago, involving this ‘std’ clash. I think it was triggered by enums, and it was so obscure, I’m not surprised it still occurs.

I am inclined to close this PR without merging - thanks for looking into it, but the current macro system works well, and I think time improving test coverage could be better spent elsewhere.

**Comment by StephanTLavavej** [comment]
For the record, the bug I filed was Dev10-663834 "The compiler emits `"error C2872: 'std' : ambiguous symbol"` even though the STL carefully says `::std::`". I could dig up the exact repro and file it again, if we're ever interested in testing this scenario.

**Comment by StephanTLavavej** [comment]
That bug was fixed at some point in the last 11 years, so never mind!

**Comment by StephanTLavavej** [comment]
There was a second compiler bug here, MS-internal VSO-1127764, caused by the compiler emitting overloads for `std::align_val_t` without full qualification; I figured out how to fix it.

**Pull request closed without merging**